### PR TITLE
Fix rere related failing replication

### PR DESCRIPTION
**Pull request by alexjg**
Introduce a test case which reproduces a strange failure to replicate which only occurs when `rere` is enabled with the patched libgit which respects force flags. Issue to describe the problem in more detail incoming.

### CODE REVIEW

**Comment by alexjg** [comment]
The fix was in the patch of libgit that we're using. I've left the test in because I think it's useful to have some coverage of this path - it being a somewhat common one.

**Review by FintanH — APPROVED** [review]
Glad you found the issue :) I retried the test suite because it was daemon tests failing.

File: test/src/test/integration/librad/scenario/tracked_references.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// Copyright © 2019-2020 The Radicle Foundation <hello@radicle.foundation>
```
Comment (xla) [inline test/src/test/integration/librad/scenario/tracked_references.rs:1]: This is the old license header, let's stick to the updated one for all new files.

File: test/src/test/integration/librad/scenario/tracked_references.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        identities,
+        storage::ReadOnlyStorage,
+        tracking,
+        types::{Namespace, Reference},
+        util::quick_commit,
+    },
+    git_ext::tree,
+    reflike,
+};
+use std::ops::Index as _;
```
Comment (xla) [inline test/src/test/integration/librad/scenario/tracked_references.rs:21]: Minor nit, some consistency in import order would make my inner OCdaemon happy.

File: test/src/test/integration/librad/scenario/tracked_references.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+fn config() -> testnet::Config {
+    testnet::Config {
+        num_peers: nonzero!(2usize),
+        min_connected: 2,
+        bootstrap: testnet::Bootstrap::from_env(),
+    }
+}
+
+#[test]
```
Comment (xla) [inline test/src/test/integration/librad/scenario/tracked_references.rs:31]: The attribute seems a bit lost with a big block of comment between it and the function it applies to.

File: test/src/test/integration/librad/scenario/tracked_references.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+/// Peers should be able to see references created by peers in their tracking
+/// graph. To test this we do the following:
+///
+/// - Create a testnet with two peers, peer1, and peer2
```
Comment (xla) [inline test/src/test/integration/librad/scenario/tracked_references.rs:35]:
```suggestion
```

Can be assumed.

**Review by xla — COMMENTED** [review]
utACK. Seems legit. :>

File: test/src/test/integration/librad/scenario/tracked_references.rs (hunk 1 above)
Comment (alexjg) [reply test/src/test/integration/librad/scenario/tracked_references.rs:1]: You're referring to this right?:

```
// Copyright © 2021 The Radicle Link Contributors
//
// This file is part of radicle-link, distributed under the GPLv3 with Radicle
// Linking Exception. For full terms see the included LICENSE file.
```

Comment (xla) [reply test/src/test/integration/librad/scenario/tracked_references.rs:1]: yus

File: test/src/test/integration/librad/scenario/tracked_references.rs (hunk 2 above)
Comment (alexjg) [reply test/src/test/integration/librad/scenario/tracked_references.rs:21]: Interesting, what consistency are you referring to? `rust-analyzer` arranged these imports in alphabetical order, are you thinking of something like nested imports appearing below non-nested imports?

Comment (xla) [reply test/src/test/integration/librad/scenario/tracked_references.rs:21]: Dropping an example which show-cases order: std, thrid-party crates, first-party crates, super, crate:

https://github.com/radicle-dev/radicle-link/blob/33c0db4402fb81c1e10bd7742531d972afc98379/librad/src/net/protocol.rs#L6-L33

Comment (alexjg) [reply test/src/test/integration/librad/scenario/tracked_references.rs]: Thanks! Interestingly you have to have a new line between each category or `cargo fmt` will just silently reorder everything to alphabetical order for you.

**Review by xla — APPROVED** [review]
↩️ 🌝 📖 🎌

**Pull request merged**